spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/filter/factory/SpringCloudCircuitBreakerFilterFactory.java

```diff
@@ -99,15 +99,16 @@ public GatewayFilter apply(Config config) {
 				return cb.run(chain.filter(exchange).doOnSuccess(v -> {
 					if (statuses.contains(exchange.getResponse().getStatusCode())) {
 						HttpStatus status = exchange.getResponse().getStatusCode();
-						exchange.getResponse().setStatusCode(null);
-						reset(exchange);
 						throw new CircuitBreakerStatusCodeException(status);
 					}
 				}), t -> {
 					if (config.getFallbackUri() == null) {
 						return Mono.error(t);
 					}
 
+					exchange.getResponse().setStatusCode(null);
```

Comment:
This concerns me a bit. Just want to make sure. Since we `throw new CircuitBreakerStatusCodeException(status)`, resetting the response still happens because this is fallback part of `cb.run()`?

Reply:
I also dont understand why the code was removed from above

Reply:
> This concerns me a bit. Just want to make sure. Since we throw new CircuitBreakerStatusCodeException(status), resetting the response still happens because this is fallback part of cb.run()?

Yes right. And also, I think that if other Exception(not only `CircuitBreakerStatusCodeException` but also all of Exception) occurs and fallback exists, we need to reset response status code and exchange.

Reply:
But why do we not reset the response status in the above case any longer?

Reply:
As @spencergibb said, because we `throw new CircuitBreakerStatusCodeException(status)`, resetting the response status still happens in fallback part.